scripts/lint

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -e
+
+echo
+echo " -- CHECKING TYPES WITH PYRIGHT --"
+echo
+
+scripts/pyright pystac tests
+
+echo
+echo " -- LINTING WITH FLAKE8 --"
+echo
+
+flake8 pystac tests
+
+echo
+echo " -- CHECKING FORMAT WITH BLACK --"
+echo
+
+black --check pystac
+black --check tests
+
+echo
+echo " -- CHECKING SPELLING WITH CODESPELL --"
+echo
+
+# Code spelling
+codespell -I .codespellignore -f \
+    pystac/*.py pystac/**/*.py \
```

Comment:
Does `codespell` do its own glob expansion? If not you probably want to run `shopt -s globstar` before this line or use the `git ls-files` approach outlined above.

Reply:
I actually just copied this over from the `scripts/test` file, so I'm not totally sure. @lossyrob Do you have any insight on this?

Reply:
[This commit](https://github.com/stac-utils/pystac/pull/401/commits/97bed90378e32ebf08e3a0c80f004cf20d1d8d09) should solve this issue.

Reply:
Sounds good. I'm going to keep this as-is in this PR and leave it to #401 to introduce the fix.